components/LocationSelector/LocationSelector.vue

```diff
@@ -102,7 +102,14 @@ import { BAccordion } from 'bootstrap-vue-next';
 export default {
   name: 'LocationSelector',
   components: { SelectorBase, BedPicker, BAccordion },
-  emits: ['error', 'ready', 'update:selected', 'update:beds', 'valid'],
+  emits: [
+    'error',
+    'ready',
+    'update:selected',
+    'update:beds',
+    'update:allBeds',
```

Comment:
Instead of emitting this new event, have the `LocationSelector` internally set the beds that are selected by default. To do this:
- add a `prop` `selectAllBedsByDefault` which defaults to `false`.
  - when this `prop` is `true` all beds will be selected when the `BedPicker` is first shown.
    - You should be able to use the `beds` computed property to initialize the `checkedBeds` array. 
  - when this `prop` is `false` no beds will be selected when the `BedPicker` is first shown.

This should then also be tested in the `LocationSelector` tests and the tests for the `allBeds` event should be removed.